crates/rolldown/tests/fixtures/format/iife/basic/artifacts.snap

```diff
@@ -12,14 +12,14 @@ input_file: crates/rolldown/tests/fixtures/format/iife
 
 
 //#region main.js
-Promise.resolve().then(function() {
+(Promise.resolve()).then(function() {
```

Comment:
This newly added paren is unexpected, we could merge it first, could you open a new issue in `oxc` ? 
https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&corejs=3.21&spec=false&loose=false&code_lz=AoJw9gtglgzgpgOhHGYA2A3OAKAlAgFwAs4A7bAMwFdSBjAqMc3AAgG8AoASGQKpFIsA-kIJgAogGUAstmQBHKlGRCKYMHlwBuDgF9tHIA&debug=false&forceAllTransforms=false&modules=false&shippedProposals=false&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env&prettier=false&targets=Electron-1.8%252CNode-10.13&version=7.24.10&externalPlugins=&assumptions=%7B%7D

Reply:
Caused by https://github.com/oxc-project/oxc/pull/4245#issue-2407249521

I will send an issue for this later